### ISSUE TITLE
SPA proxying breaks with double-slash at beginning of path

### ISSUE THREAD

**Issue by jez9999** [dotnet/aspnetcore#43662]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When running an ASP.NET SPA site which uses `.UseProxyToSpaDevelopmentServer()`, the following kind of URL breaks the proxying:

```
https://localhost:44381//foo/bar
```

This is because internally, `SpaProxy` is constructing a `Uri` like this:

```
var targetUri = new Uri(
            baseUri,
            context.Request.Path + context.Request.QueryString);
```

If the second argument begins `//`, `Uri` treats it as an absolute path and throws away `baseUri`.  A period `.` needs to be prefixed to the relative URI to avoid this behaviour.

### Expected Behavior

The URL will proxy successfully and not break.

### Steps To Reproduce

Setup an ASP.NET SPA site, and try to access any path on the local server beginning with a double-slash.  eg. `https://localhost:44381//foo/bar`

### Exceptions (if any)

```
System.Net.Http.HttpRequestException

Failed to proxy the request to http://aa/b, because the request to the proxy target failed. Check that the proxy target server is running and accepting requests to http://localhost:3000/.\n\nThe underlying exception message was 'No such host is known. (aa:80)'.Check the InnerException for more details.
```

### .NET Version

5.0.411

**Comment by javiercn** [comment]
@jez9999 thanks for contacting us.

.NET 5.0 is out of support, as per our support policy we will recommend that as a first step you update to .NET 6.0. We have changed the approach we use for SPAs in .NET 6.0 to use the SPA proxy in front of the ASP.NET backend as that's the canonical approach SPA frontends recommend.

**Comment by jez9999** [comment]
@javiercn Is having the ASP.NET app in front of the SPA app now a completely unsupported use-case?  That's really quite annoying if so.

**Comment by Tratcher** [comment]
@blowdart says we need to fix this regardless of UseProxyToSpaDevelopmentServer's future status.

**Comment by blowdart** [comment]
Thank you :)

**Comment by jez9999** [comment]
I submitted a PR for this by the way, #43663

**Comment by danroth27** [comment]
> @javiercn Is having the ASP.NET app in front of the SPA app now a completely unsupported use-case? That's really quite annoying if so.

HI @jez9999. Using UseProxyToSpaDevelopmentServer is still supported, but can you tell us more why hosting the SPA dev server in front of the ASP.NET Core app is problematic? We believe that is the standard setup for most folks doing frontend development with JS based frontend. Are there things that block you from moving to the SpaServices.Extensions model to the new one?

**Comment by jez9999** [comment]
I don't like the "black magic" of using `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` to inject the `SpaProxy` module - which could potentially change at any time - in order that some middleware is registered to support the new technique.  I liked explicitly specifying all my middleware in a Startup, or the Program, class.  It wouldn't be so bad if there were a way to use the new technique without the whole "some package registers mystery middleware" thing.

**Comment by danroth27** [comment]
@jez9999 Got it. Thanks for sharing this feedback!

**Comment by varsi94** [comment]
@jez9999 This should be fixed with #43407.
